app/services/flow/flow_state_machine.rb

```diff
@@ -179,6 +179,16 @@ def analytics_properties
         irs_reproofing: effective_user&.decorate&.reproof_for_irs?(
           service_provider: current_sp,
         ).present?,
+      }.merge(native_camera_ab_testing_variables)
+    end
+
+    def native_camera_ab_testing_variables
```

Comment:
Granted that FlowStateMachine is a bit of a lost cause in terms of its intended purpose of being generally reusable, but would it make sense to consider moving extra analytics like this out of the base module and into specific implementation? For example, maybe `DocAuthController` (and other relevant controllers) should have an `extra_analytics_properties` method, and FlowStateMachine can check if the controller in which it's included responds to that method?

Reply:
do we need these fields on every single step? or are the only relevant on certain steps?

Reply:
Added `extra_analytics_properties` to the base flow (returning `{}`) then limited the test logging to the doc_auth and capture_doc flows in 3da71380c

We don't technically need this in every step, but we are doing this at all because the initial logging was inadequate, and the FSM didn't have a mechanism for individual steps to log extra params. I could add that, but don't really want to invest any time into improving code we are actively planning on removing.

Reply:
Also, the native camera a/b test is just that, at test, and will be removed once we get the info from the test.